yogstation/code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -66,6 +66,10 @@
 	if(alert("WARNING: Are you sure you wish to self-destruct? This action cannot be undone!",,"Yes","No") != "Yes")
 		return
 
+	if(usr.stat == DEAD)
+		to_chat(usr, "<span class='danger'>You are already dead.</span>")
+		return //won't work if dead
```

Comment:
Seems silly to check for dead twice, why not just move the alert above the check?

Reply:
If I removed the top check, the alert box would show up even if the borg was dead, it just wouldn't do anything.  By checking before, we avoid asking people to confirm actions that we already know they can't do.  I think it's cleaner, and the performance impact is hilariously negligible.